src/main/java/com/sinthoras/visualprospecting/gui/xaeromap/waypoints/WaypointManager.java

```diff
@@ -0,0 +1,49 @@
+package com.sinthoras.visualprospecting.gui.xaeromap.waypoints;
+
+import com.sinthoras.visualprospecting.Tags;
+import com.sinthoras.visualprospecting.gui.model.SupportedMods;
+import com.sinthoras.visualprospecting.gui.model.layers.WaypointProviderManager;
+import com.sinthoras.visualprospecting.gui.model.waypoints.Waypoint;
+import xaero.common.minimap.waypoints.WaypointsManager;
+
+import java.util.Hashtable;
+
+public class WaypointManager extends com.sinthoras.visualprospecting.gui.model.waypoints.WaypointManager {
+
+	private static final Hashtable<Integer, xaero.common.minimap.waypoints.Waypoint> xWaypointTable = WaypointsManager.getCustomWaypoints(Tags.MODID);
+
+	private WaypointWithDimension xWaypoint;
+	private final WaypointTypes id;
```

Comment:
id is a bad name, since it is not an individual identifier for a waypoint. Just call it `waypointType`